x-pack/plugins/security/public/account_management/api_keys_page/api_keys_table.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+// @ts-ignore
+import { formatDate } from '@elastic/eui/lib/services/format';
+import {
+  EuiBadge,
+  EuiButtonEmpty,
+  EuiHideFor,
+  EuiInMemoryTable,
+  EuiInMemoryTableProps,
+  EuiShowFor,
+  EuiText,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import { ApiKey } from '../../../common/model';
+
+export type ApiKeysTableProps = Omit<EuiInMemoryTableProps<ApiKey>, 'columns'> & {
+  createdItemId?: ApiKey['id'];
+  onInvalidateItem(item: ApiKey): void;
+};
+
+export const ApiKeysTable: FunctionComponent<ApiKeysTableProps> = ({
+  createdItemId,
+  onInvalidateItem,
+  ...props
+}) => {
+  const actions = [
+    {
+      render: (item: ApiKey) => (
+        <>
+          <EuiShowFor sizes={['xs', 's']}>
+            <EuiButtonIcon
+              iconType="minusInCircle"
+              color="danger"
+              disabled={props.loading}
+              onClick={() => {
+                onInvalidateItem(item);
+              }}
+            />
+          </EuiShowFor>
+          <EuiHideFor sizes={['xs', 's']}>
+            <EuiButtonEmpty
+              iconType="minusInCircle"
+              color="danger"
+              flush="right"
+              disabled={props.loading}
+              onClick={() => {
+                onInvalidateItem(item);
+              }}
+            >
+              <FormattedMessage
+                id="xpack.security.accountManagement.apiKeys.invalidateButton"
+                defaultMessage="Invalidate"
+              />
+            </EuiButtonEmpty>
+          </EuiHideFor>
+        </>
+      ),
+    },
+  ];
+
+  return (
+    <EuiInMemoryTable
+      {...props}
+      items={props.loading && !createdItemId ? [] : props.items}
```

Comment:
Adding a `tableCaption` would be a nice touch for screen readers:

```suggestion
      items={props.loading && !createdItemId ? [] : props.items}
      tableCaption={i18n.translate('xpack.security.accountManagement.apiKeys.tableCaption', {
              defaultMessage: 'Below is a table of your API Keys',
            })}
```